MetaMask/metamask-mobile
# Sent transaction icon changes incorrectly when the language changes

**Description**
There was more than one issue, one of them was when we change the language, the icon of a transaction was not consistent, a sent icon was changed for one swipe icon. 
Also, we were comparing translations, and when we change the language, some of them stayed on the past language, because the component would re-render, but some util function was compiling faster than the new language rendered.

**Proposed Solution**
Comparing the action key and not the action key translation.
 [Technical] - Refactored logic for we do not compare translations anymore.

**Code Impact**
Low/Medium - Will affect every info of transactions, on TransactionsView and Asset.

**Test Cases**
Case 1:
 * Switch to Polygon Mainnet
 * Do a swap to USDT
 * Open Activity View
 * Open Asset View
 * Change language to Spanish
 * The icon and the translation should mean the same
Case 2:
 * Switch to Polygon Mainnet
 * Send a token to yourself 
 * Open Activity View
 * Open Asset View
 * Change language to Spanish
 * The icon and the translation should mean the same
Case 3:
 * Switch to Polygon Mainnet
 * Send MATIC to another wallet 
 * Open Activity View
 * Open Asset View
 * Change language to Spanish
 * The icon and the translation should mean the same
Case 4:
 * Switch to Polygon Mainnet
 * Send another token (Like USDT) to yourself and to another wallet
 * Open Activity View
 * Open Asset View
 * Change language to Spanish
 * The icon and the translation should mean the same
 Case 5:
 * Switch to rinkeby
 * Import a NFT
 * Send it to other wallet
 * Open Activity View
 * Change language to Spanish
 * The icon and the translation should mean the same

**Some expected English translations of transactions**
<img width="287" alt="image" src="https://user-images.githubusercontent.com/46944231/186048738-14ea9f43-55ba-4661-82de-035bd0dee8ac.png">


**Screenshots/Recordings**
https://recordit.co/eQPcABRTaA
<div display:'inline'>
<img src='https://user-images.githubusercontent.com/46944231/186047378-75325f2a-1a16-4e3f-89d7-16df9a210fe4.png' width=250 />
<img src='https://user-images.githubusercontent.com/46944231/186047386-b9583cda-cf1a-4a98-a511-c94ff7443145.png' width=250 />
</div>
<div display:'inline'>
<img src='https://user-images.githubusercontent.com/46944231/186047395-a5bd4b31-9588-45d4-a2a5-6d4f59a31347.png' width=250 />
<img src='https://user-images.githubusercontent.com/46944231/186047397-e5df9fa5-372c-4c62-9556-17b84ce030d2.png' width=250 />
</div>

**Issue**
https://github.com/MetaMask/metamask-mobile/issues/4814
Progresses #4184 

**Checklist**
